### PR TITLE
Clean up header file inclusion, sequel of c4f28da

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
* Use "top-down" header file inclusion ordering
* Add missing header files
* Remove useless header files
    
It should be possible to include header files in any order, but we're probably not there yet.

Also, remove a `BLOSC_EXPORT` from a source file, it should be used in the header file when declaring the function.

### CODE REVIEW

File: plugins/codecs/codecs-registry.c [hunk 1]
```diff
@@ -4,46 +4,47 @@
...
 void register_codecs(void) {
 
   blosc2_codec ndlz;
   ndlz.compcode = BLOSC_CODEC_NDLZ;
   ndlz.version = 1;
   ndlz.complib = BLOSC_CODEC_NDLZ;
-  ndlz.encoder = ndlz_compress;
-  ndlz.decoder = ndlz_decompress;
+  ndlz.encoder = &ndlz_compress;
+  ndlz.decoder = &ndlz_decompress;
```
Comment (FrancescAlted) [inline plugins/codecs/codecs-registry.c:20]: This is a serious fix.  I wonder why it was working before (?).

**Review by FrancescAlted — COMMENTED** [review]
This is a great PR.  I just have put a question, in case you can shed some light on it.

File: plugins/codecs/codecs-registry.c (hunk 1 above)
Comment (DimitriPapadopoulos) [reply plugins/codecs/codecs-registry.c:20]: Actually, they are equivalent. A function does not have a “value” so C compilers always interpret that kind of expressions as a pointer to the function. From the C standard:
> A _function designator_ is an expression that has function type. Except when it is the operand of the `sizeof` operator, the `_Alignof` operator, or the unary `&` operator, a function designator with _type_ ‘‘function returning type’’ is converted to an expression that has type ‘‘pointer to function returning _type_’’.

I think most C style manuals recommend the `&` to stress we store a pointer to the function, but it is not compulsory. I seem to recall the '&' is usually dropped in C++.

Comment (FrancescAlted) [reply plugins/codecs/codecs-registry.c:20]: Cool!  Makes sense now.

**Review by FrancescAlted — APPROVED** [review]

**Comment by FrancescAlted** [comment]
Thanks @DimitriPapadopoulos !

**Pull request merged**